backend/src/api/game/challenge.rs

```diff
@@ -19,34 +68,107 @@ pub struct NewChallengeRequest {
     // cannot be played first.
     pub tournament_queen_rule: bool,
 
+    // The challenger's color choice
+    pub color_choice: ColorChoice,
+
     pub game_type: GameType,
 }
 
 #[derive(Serialize)]
-pub struct NewChallengeResponse {
-    challenge_url: String,
+#[serde(rename_all = "camelCase")]
+pub struct GameChallengeResponse {
+    pub id: Uuid,
+    pub challenger: User,
+    pub game_type: GameType,
+    pub ranked: bool,
+    pub public: bool,
+    pub tournament_queen_rule: bool,
+    pub color_choice: ColorChoice,
+    pub created_at: DateTime<Utc>,
+}
+
+impl GameChallengeResponse {
+    pub async fn from_model(challenge: &GameChallenge, pool: &DbPool) -> Result<Self, ServerError> {
+        let challenger = match challenge.get_challenger(&pool).await {
+            Ok(challenger) => challenger,
+            Err(diesel::result::Error::NotFound) => {
+                let uid = challenge.challenger_uid.clone();
+                return Err(ChallengeError::MissingChallenger(uid).into());
```

Comment:
There would be something super wrong here right? We need to make sure we create an error class for these errors that are important for us vs just something that the user fucked up

Reply:
yup! and `MissingChallenger` is mapped to a 500 error, to indicate that the user didn't do anything wrong, this is a bug on our end